docs/src/proposals/epoch-boundary-stake-reward.md

```diff
@@ -0,0 +1,26 @@
+---
+title: Stake Reward At Epoch Boundary
+---
+
+## Problem
+
+With the increase of number of stake accounts, computing and redeeming the stake rewards at the start slot of the epoch boundary becomes very expensive. Currently, with 550K stake accounts, the stake reward time has already taken more than 10 seconds. This prolonged computation slows down the network, and can cause large number of forks at the epoch boundary, which make the matter even worse.
+
+## Proposed Solutions
+
+Instead of computing and reward stake accounts at epoch boundary, we will decouple reward computation and reward credit for stake accounts. At the start of the new epoch, a background thread is forked to compute the rewards. At slot `N` after the epoch, the rewards will be credited to all stake accounts. We could call these `epoch_start..epoch_start+N` slots are `rewarding interval`.  `N` is chosen to be sufficiently large so that the background computation should have completed and the result of the reward computation is available at the end of `rewarding interval`. `N` can be fixed such as 100 (roughly equivalent to 50 seconds), or chosen as a function of the number of stake accounts, `f(num_stake_accounts)`. When reached slot `N`, the bank will fetch the reward computation results from the background thread, and credit the rewards to all the stake accounts.
```

Comment:
However N is chosen, if we skip slot N, we have to redo all the rewards stores into slot N+1...
This gets me thinking about how expensive the stores are for 500k(?) writes in 1 slot.